### ISSUE TITLE
Video freeze at 6 hours

### ISSUE THREAD

**Issue by kingsombra3265** [anxdpanic/plugin.video.youtube#1007]
* Add-on Version: 7.1.1.3
* Kodi Version: 21.1
* Kodi GUI Language: English
* Operating System: Android 12 a7m
* Operating System Language: English

------

### Expected Behavior

When playing a video that is over 6 hours long it continues to play to the end.

------

### Current Behavior

Video freezes at 6 hours and a short sound loop can be heard over and over.

------

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Play a video that is over 6 hours in length
2. Wait until the 6 hour mark

------

### Log

https://paste.kodi.tv/jazojemoho.kodi

------

### Additional Information

I found the following particularly interesting:

AddOnLog: inputstream.adaptive: [AS-11] Download finished: https://rr5---sn-ab5l6nkd.googlevideo.com/videoplayback?expire=1733142763&ei=i1RNZ7j2FoCBkucP_pyj4AI&ip=74.70.117.158&id=o-ADyJDOGQbpiqfDQSlTfNutvM8vgBUB83tVRgN2U90zAn&itag=140&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&met=1733121163%2C&mh=2s&mm=31%2C29&mn=sn-ab5l6nkd%2Csn-ab5sznze&ms=au%2Crdu&mv=m&mvi=5&pl=18&rms=au%2Cau&initcwndbps=1725000&bui=AQn3pFQ3ig5I7TE8_Wrq11LAUYJQUTVzpBRtvadiwz9sQWgNHdd9qIv0apq0u9fHRVMYQPsCZf02fR22&vprv=1&svpuc=1&mime=audio%2Fmp4&rqh=1&gir=yes&clen=465411726&dur=28757.680&lmt=1708868493684285&mt=1733120709&fvip=2&keepalive=yes&fexp=51326932%2C51335594&c=ANDROID_VR&txp=4532434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIhAK_JDLI6Sz57347WKQz88n0HQf8OPQO6RSwkxR3mJ4_uAiBj8ycyVBMMGFvs1AUF-aQQvVs-a7ho0OB63fzTysFbPg%3D%3D&lsparams=met%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=AGluJ3MwRQIgeBBlDSK3VMlwEE1_odLMvbsaxAkICmkuhkSZ5YzitrMCIQDomf8TrOIHn4OtpqRMKKg2x2mU-huVi4mpBmLOFdXLlQ%3D%3D&range=0-465411726 (downloaded 161577 byte, speed 4382581.00 byte/s)

Note it says something about "expire".

Perhaps youtube does have a time limit after all.

------

**Comment by MoojMidge** [comment]
Can you try again with the log? The debug logging needs to be enabled before the video playback starts and extend until 5 minutes before the playback fails. Everything else can be cut from the middle.

**Comment by kingsombra3265** [comment]
I redid my edits of the log.

I used regular expressions to remove "WEATHER: Downloading weather" lines and "WARNING ActiveAE - large audio sync error" which I felt were kind of not needed.

I also removed a section that is not as big from the middle.

I went from 2am which was 30 minutes after the video start, to 7am which was 30 minutes before the end.

The result is here:

https://paste.kodi.tv/eyakejihax.kodi

**Comment by MoojMidge** [comment]
Thanks. The stream links are only valid for 6 hours and aren't updated automatically based on segment time base. Will probably need a new player request to refresh the stream links in additional periods in the manifest, but I'm not sure how InputStream.Adaptive supports this.

The problem is clear, but don't think there is a workable solution at the moment.

**Comment by kingsombra3265** [comment]
I don't mind too much yet. I've mainly been using freedownloadmanager + elephant plugin to download the videos to my NAS drive, that I really want and can play them that way. (although this does make the addon a little less useful for me)

Do you think I should drop a small note over in "Support > Add-on Support > VideoPlayer InputStream" pointing to this ticket #1007?

Edit: One thing I think I am going to do as a workaround, is to completely turn off InputStream.Adaptive and make the maximum resolution 720p. Having a maximum of 720p is actually a preference of mine anyway.

**Comment by MoojMidge** [comment]
> Do you think I should drop a small note over in "Support > Add-on Support > VideoPlayer InputStream" pointing to this ticket #1007?

Not at this stage

> One thing I think I am going to do as a workaround, is to completely turn off InputStream.Adaptive and make the maximum resolution 720p. Having a maximum of 720p is actually a preference of mine anyway.

You can try but it won't help. The issue is not with InputStream.Adaptive and I'm pretty sure that disabling it will prevent most videos from playing correctly.

Since you are creating strm files, one thing you could do is change your strm files to something like this
```
plugin://plugin.video.youtube/play?video_id=kT-f2mfNL7s&start=0&end=21600
plugin://plugin.video.youtube/play?video_id=kT-f2mfNL7s&start=21600
```

**Comment by kingsombra3265** [comment]
UG. OK. Kodi has never been known for good continuous play and doing that would introduce a nice little annoying break that the 6 hour mark.

I guess Ill just have to continue downloading the larger video's that I want to use for ambience while I sleep.

**Comment by MoojMidge** [comment]
Yes there will be a gap, but note that if this ever gets fixed in a way that doesn't require manually splitting up videos in a playlist, then it is almost certain that there will still be a gap that results from having to retrieve the updated video streams.

Anyway, the use of clips in a playlist/strm file doesn't actually work at the moment. Added a fix for this which will be available in the next beta release, if you do end up using that as a workaround.

**Comment by MoojMidge** [comment]
This test version includes a workaround for this problem:
https://github.com/MoojMidge/plugin.video.youtube/archive/refs/heads/v7.5.zip

You will need to re-enable InputStream.Adaptive and MPEG-DASH (or better, run the Setup Wizard, to configure the addon correctly).

Can you test and confirm whether this works?

**Comment by kingsombra3265** [comment]
I am sorry to say that at this point, I have abandoned the plugin all together.

As a matter of principle, I will not use it if it requires that I get a google account.

In its place I am using smarttube.

**Comment by MoojMidge** [comment]
Suit yourself.

The addon doesn't require a google account any more than SmartTube does, but let's just leave it here.
